File: Examples2018/DataManagementFileInternalMemory/app/src/main/java/course/examples/datamanagement/fileinternal/InternalFileWriteReadActivity.java
package course.examples.datamanagement.fileinternal;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;

public class InternalFileWriteReadActivity extends Activity {

	private final static String fileName = "TestFile.txt";

	@Override
	public void onCreate(Bundle savedInstanceState) {

		super.onCreate(savedInstanceState);

		setContentView(R.layout.main);
		TextView textView = findViewById(R.id.textview);

		String TAG = "IntFileWriteRead";

		// Check whether fileName already exists in directory used
		// by the openFileOutput() method.
		// If the text file doesn't exist, then create it now


		if (!getFileStreamPath(fileName).exists()) {
			try {
				writeFile();
			} catch (FileNotFoundException e) {
				Log.i(TAG, "FileNotFoundException");
			}
		}

		
		// Read the data from the text file and display it
		try {
			
			readFileAndDisplay(textView);

		} catch (IOException e) {
			Log.i(TAG, "IOException");
		}
	}

	private void writeFile() throws FileNotFoundException {

		FileOutputStream fos = openFileOutput(fileName, MODE_PRIVATE);

		PrintWriter pw = new PrintWriter(new BufferedWriter(
				new OutputStreamWriter(fos)));

		pw.println("Line 1: This is a test of the File Writing API");
		pw.println("Line 2: This is a test of the File Writing API");
		pw.println("Line 3: This is a test of the File Writing API");

		pw.close();

	}

	private void readFileAndDisplay(TextView tv) throws IOException {

		FileInputStream fis = openFileInput(fileName);
		BufferedReader br = new BufferedReader(new InputStreamReader(fis));

		String line;
		String sep = System.getProperty("line.separator");

		while (null != (line = br.readLine())) {
			tv.append(line + sep);
		}

		br.close();

	}

}

File: ExamplesKotlin/DataManagementFileInternalMemory/app/src/main/java/course/examples/datamanagement/fileinternal/InternalFileWriteReadActivity.kt
package course.examples.datamanagement.fileinternal

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.util.Log
import android.widget.TextView

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.FileNotFoundException
import java.io.IOException
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.io.PrintWriter

class InternalFileWriteReadActivity : Activity() {

    companion object {
        private const val TAG = "IntFileWriteRead"
        private const val FILE_NAME = "TestFile.txt"
    }

    public override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)

        setContentView(R.layout.main)
        val textView = findViewById<TextView>(R.id.textview)


        // Check whether FILE_NAME already exists in directory used
        // by the openFileOutput() method.
        // If the text file doesn't exist, then create it now

        if (!getFileStreamPath(FILE_NAME).exists()) {
            try {
                writeFile()
            } catch (e: FileNotFoundException) {
                Log.i(TAG, "FileNotFoundException")
            }

        }


        // Read the data from the text file and display it
        try {

            readFileAndDisplay(textView)

        } catch (e: IOException) {
            Log.i(TAG, "IOException")
        }

    }

    @Throws(FileNotFoundException::class)
    private fun writeFile() {

        val fos = openFileOutput(FILE_NAME, Context.MODE_PRIVATE)

        val pw = PrintWriter(BufferedWriter(OutputStreamWriter(fos)))

        pw.println("Line 1: This is a test of the File Writing API")
        pw.println("Line 2: This is a test of the File Writing API")
        pw.println("Line 3: This is a test of the File Writing API")

        pw.close()

    }

    @Throws(IOException::class)
    private fun readFileAndDisplay(tv: TextView) {

        val sep = System.getProperty("line.separator")
        val fis = openFileInput(FILE_NAME)
        val br = BufferedReader(InputStreamReader(fis))

        br.forEachLine {
            tv.append(it + sep)
        }
        br.close()
    }
}